contrib/epee/src/readline_buffer.cpp

```diff
@@ -167,30 +165,42 @@ static int handle_enter(int x, int y)
   char* line = NULL;
 
   line = rl_copy_text(0, rl_end);
+  std::string test_line = line;
+  boost::trim_right(test_line);
+
   rl_crlf();
   rl_on_new_line();
+
+  if(test_line.empty())
+  {
+    last_line = "";
+    rl_set_prompt(last_prompt.c_str());
+    rl_replace_line("", 1);
+    rl_redisplay();
+    have_line.notify_one();
+    return 0;
+  }
+
   rl_set_prompt("");
   rl_replace_line("", 1);
   rl_redisplay();
   
-  std::string test_line = line;
-  boost::trim_right(test_line);
-  if (test_line.length() > 0)
+  if (!test_line.empty())
   {
     last_line = test_line;
```

Comment:
The ever optimizer in me is thinking about a `std::move` with a last_line.c_str() on the next line.